lib/msf/core/post/windows/priv.rb

```diff
@@ -124,23 +124,19 @@ def is_system?
   # if running on a system that does not have UAC
   #
   def is_uac_enabled?
-    uac = false
-    winversion = session.sys.config.sysinfo['OS']
+    return false if is_system?
 
-    if winversion =~ /Windows (Vista|7|8|2008|2012|10)/
-      unless is_system?
-        begin
-          enable_lua = registry_getvaldata(
-              'HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System',
-              'EnableLUA'
-          )
-          uac = (enable_lua == 1)
-        rescue Rex::Post::Meterpreter::RequestError => e
-          print_error("Error Checking if UAC is Enabled: #{e.class} #{e}")
-        end
-      end
-    end
-    return uac
+    winversion = session.sys.config.sysinfo['OS']
+    return false unless winversion =~ /Windows (Vista|7|8|10|2008|2012|2016)/
+
+    enable_lua = registry_getvaldata(
+      'HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System',
+      'EnableLUA'
+    )
+    (enable_lua == 1)
```

Comment:
Should this be `return (enable_lua == 1)`?

Reply:
Nope.

```ruby
# irb
2.3.0 :001 > def asdf
2.3.0 :002?>   enable_lua = 1
2.3.0 :003?>   (enable_lua == 1)
2.3.0 :004?>   end
 => :asdf 
2.3.0 :005 > asdf
 => true 
2.3.0 :006 > 
```

```ruby
# irb
2.3.0 :001 > def asdf
2.3.0 :002?>   enable_lua = 0
2.3.0 :003?>   (enable_lua == 1)
2.3.0 :004?>   end
 => :asdf 
2.3.0 :005 > asdf
 => false 
2.3.0 :006 > 
```

Reply:
I just noticed this would be the last statement (if no error happens). Never mind 👍